### PR TITLE
feat(truncation): add automatic tooltip on truncated text

### PR DESCRIPTION
**Pull request by QuintonJason**
# Description

Add automatic tooltip behavior when text is truncated in Pine components. Previously, truncated text (`text-overflow: ellipsis`) had no way to reveal the hidden content. Now, hovering or focusing a truncated element shows a tooltip with the full text — but only when the content is actually overflowing.

### What changed

- **New shared utility** (`truncation-tooltip.ts`) — handles overflow detection (`scrollWidth > clientWidth`), `ResizeObserver` for dynamic re-evaluation, hover/focus listeners, and a Floating UI-positioned tooltip portal on `document.body`. Styled with Pine design tokens to match `pds-tooltip`.
- **`pds-text`** — automatically shows tooltip on hover/focus when `truncate` is set and text overflows
- **`pds-table-cell`** — same behavior; also makes `max-width` configurable via `--pds-table-cell-truncate-max-width` CSS custom property (default: `100px`)
- **`pds-copytext`** — same behavior, uses `value` prop as tooltip content

No new props required — the tooltip is automatic when `truncate` is set. No breaking changes.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

- [x] unit tests
- [x] tested manually

12 new tests for the shared utility covering: tooltip creation on overflow, no tooltip when text fits, hover/focus interactions, hover+focus overlap, empty text edge case, and cleanup. Component spec files updated with integration tests for all three components.

**Test Configuration**:

- Pine versions: 3.17.0 (current main)
- OS: macOS

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Design has QA'ed and approved this PR

### CODE REVIEW

**Comment by QuintonJason** [comment]
@coderabbitai review this pr

**Review by pixelflips — COMMENTED** [review]
Overall this is a well-implemented feature with solid architecture and good test coverage. A few accessibility and robustness items to address before merge — see inline comments.

File: libs/core/src/utils/truncation-tooltip.ts [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    .${TOOLTIP_CLASS}--visible .${TOOLTIP_CLASS}__content {
+      opacity: 1;
+      visibility: visible;
+    }
+  `;
+  document.head.appendChild(style);
+}
+
+/**
+ * Checks if an element's content is overflowing horizontally (i.e. text is truncated).
```
Comment (pixelflips) [inline libs/core/src/utils/truncation-tooltip.ts:75]: **Accessibility: Missing `aria-describedby` association**

The tooltip sets `role="tooltip"` and `aria-hidden="false"`, but there is no `aria-describedby` linking the host element to the tooltip. Without this, screen readers won't announce the tooltip content when the truncated element receives focus.

Suggestion:
- Generate a unique `id` for each tooltip portal (e.g., `pds-truncation-tooltip-${counter++}`)
- Set `aria-describedby="<tooltip-id>"` on `hostEl` when the tooltip is shown
- Remove `aria-describedby` from `hostEl` when the tooltip is hidden

This is a standard pattern per [WAI-ARIA tooltip](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/) and would make the feature properly accessible.

File: libs/core/src/utils/truncation-tooltip.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+        const padding = 5;
+
+        // Ensure tooltip stays within viewport bounds
+        const left = Math.max(padding, Math.min(rect.left, window.innerWidth - portalRect.width - padding));
+        const top = Math.max(padding, rect.top - portalRect.height - 8);
+
+        portalEl.style.left = `${left}px`;
+        portalEl.style.top = `${top}px`;
+      }
+    }
```
Comment (pixelflips) [inline libs/core/src/utils/truncation-tooltip.ts:178]: **Accessibility: Missing Escape key dismissal**

Per the [WAI-ARIA tooltip pattern](https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/), pressing `Escape` should dismiss the tooltip. This is not currently implemented.

Suggestion: Add a `keydown` listener (on `hostEl` or `document`) that calls `hideTooltip()` when `Escape` is pressed while the tooltip is visible. Something like:

```ts
function handleKeyDown(e: KeyboardEvent): void {
  if (e.key === 'Escape' && portalEl) {
    isHovering = false;
    isFocused = false;
    isHoveringPortal = false;
    cancelHideTimeout();
    removePortal();
  }
}
hostEl.addEventListener('keydown', handleKeyDown);
```

Don't forget to remove the listener in the cleanup function.

File: libs/core/src/components/pds-table/pds-table-cell/pds-table-cell.tsx [hunk 3]
```diff
@@ -74,16 +81,44 @@ export class PdsTableCell {
...
+  handleTruncateChange(newValue: boolean) {
+    if (newValue) {
+      this.initTruncationTooltip();
+    } else {
+      this.destroyTruncationTooltip();
+    }
+  }
+
   /**
    * Determines if the table is currently scrolling.
```
Comment (pixelflips) [inline libs/core/src/components/pds-table/pds-table-cell/pds-table-cell.tsx:100]: **Robustness: `textContent` may include unintended child element text**

```ts
getTooltipText: () => this.hostElement.textContent || '',
```

If a table cell contains nested elements (badges, icons with labels, nested spans, etc.), `textContent` will concatenate all of their text content, which could result in garbled or misleading tooltip text.

Consider either:
1. Using `innerText` instead (respects CSS visibility/display, closer to what the user sees), or
2. Targeting a specific text node or slotted content rather than the entire host, or
3. Documenting this as a known limitation — that `truncate` works best on cells with plain text content

**Review by pixelflips — APPROVED** [review]
LGTM!
